py4DSTEM/process/preprocess/preprocess.py

```diff
@@ -149,5 +150,117 @@ def bin_data_real(datacube, bin_factor):
         datacube.R_Nx,datacube.R_Ny,datacube.Q_Nx,datacube.Q_Ny = datacube.data.shape
         return datacube
 
+def filter_hot_pixels(
+    datacube,
+    thresh,
+    ind_compare=1,
+    return_mask=False
+    ):
+    """
+    This function performs pixel filtering to remove hot / bright pixels. We first compute a moving local ordering filter,
+    applied to the mean diffraction image. This ordering filter will return a single value from the local sorted intensity 
+    values, given by ind_compare. ind_compare=0 would be the highest intensity, =1 would be the second hightest, etc.
+    Next, a mask is generated for all pixels which are least a value thresh higher than the local ordering filter output.
+    Finally, we loop through all diffraction images, and any pixels defined by mask are replaced by their 3x3 local median.
+
+    Args:
+            datacube (DataCube):      py4DSTEM Datacube
+            thresh (float):           threshold for replacing hot pixels, if pixel value minus local ordering filter exceeds it.
+            ind_compare (int):        which median filter value to compare against. 0 = brightest pixel, 1 = next brightest, etc.
+
+        Returns:
+            datacube                     datacube              
+            mask (bool):                 (optional) the bad pixel mask
+    """
+
+    # Mean image over all probe positions
+    diff_mean = np.mean(datacube.data,axis=(0,1))
```

Comment:
in the future we might have this call a datacube fn which (a) looks to see if this has already been computed and if so returns it, else (b) computes and stores it